README.md

```diff
@@ -29,32 +29,86 @@ to the original [tmkms](https://github.com/iqlusioninc/tmkms) repository.
 
 The following signing backend providers are presently supported:
 
-#### Software-Only (not recommended; only for testing)
+### Software-Only (not recommended; only for testing)
 
 This is contained in the "providers/softsign" directory.
 
-#### Intel(R) SGX
+### Intel(R) SGX
 This is contained in the "providers/sgx" directory.
 There are two crates that need to be compiled separately:
 * `tmkms-light-sgx-app`: this is the enclave application that needs to be compiled for the `x86_64-fortanix-unknown-sgx` target, converted and signed as per [EDP instructions](https://edp.fortanix.com/docs/tasks/deployment/);
 * `tmkms-light-sgx-runner`: this is the (host) runner application that is used to load the enclave application
 and interface with the host system.
 
-##### Installation
-TODO
-##### Configuration
-TODO
-##### Running
-TODO
+#### Installation
+
+First, Setup your sgx machine with [EDP Installation guide](https://edp.fortanix.com/docs/installation/guide/)
+
+Build `tmkms-light-sgx-runner`
+```bash
+cargo build --target x86_64-unknown-linux-gnu -p tmkms-light-sgx-runner --release
+```
+Build `tmkms-light-sgx-app`
+```bash
+cargo build --target x86_64-fortanix-unknown-sgx -p tmkms-light-sgx-app --release
+```
+
+Follow [EDP instructions](https://edp.fortanix.com/docs/tasks/deployment/) for SGXS conversion and signing.
+> :warning: For SGXS conversion, change `--heap-size/--stack-size` value to `0x40000`, and `--threads 2` should be enough.
+
+
+#### Configuration
+
+- Place the generated `*.sgxs` and `*.sig` files into `tmkms/enclave` directory.
+- Place the `tmkms-light-sgx-runner` binary to `tmkms` directory.
+
+*tmkms init*
+
+Provide your chain `bech32_prefix`
+```bash
+$ tmkms-light-sgx-runner init -b bech32_prefix -p "bech32"
+```
+
+> :warning: For those who are running Azure sgx machine, one may want to run `init` command with cloud backup key.
+> In Azure, `CPU-affinity` isn’t guaranteed, so for SGX “sealing” (a way to encrypt the validator key that only a particular CPU can decrypt the validator key) can’t be fully relied on. Please follow `With cloud backup key`
+
+<details>
+  <summary>With cloud backup key</summary>
+
+One may provide flag `-e backup_key_path` which is to encrypt and decrypt `consensus-key.backup` in directory specified in `-k backup_data_path`
+
+One may also want to use [Azure Key Vault secret](https://docs.microsoft.com/en-us/azure/key-vault/secrets/quick-create-python) to store your backup key.
+
+For security reason, one should always delete the backup key and retrieve it from Azure key vault secret.
```

Comment:
```suggestion
> :warning: After storing your backup key in Azure Key Vault or its equivalents, make sure to delete the backup key from any local environments to prevent its potential leakage.
> :warning: The instructions do not cover the adequate secure policy setup in Azure Key Vault or its equivalents, so make sure the policy restricts the backup key access only to particular instances during a recovery process.
> :warning: The current cloud key backup implementation is very rudimentary and potentially dangerous -- *use at your own risk* -- we plan to enhance this implementation once solutions for "cloud sealing" become generally available.
```